source/fuzz/fuzzer.h

```diff
@@ -128,38 +129,23 @@ class Fuzzer {
   // instruction has a distinct unique id.
   bool ApplyPassAndCheckValidity(FuzzerPass* pass) const;
 
-  // Target environment.
-  const spv_target_env target_env_;
-
   // Message consumer that will be invoked once for each message communicated
   // from the library.
-  MessageConsumer consumer_;
-
-  // The initial binary to which fuzzing should be applied.
-  const std::vector<uint32_t>& binary_in_;
-
-  // Initial facts known to hold in advance of applying any transformations.
-  const protobufs::FactSequence& initial_facts_;
-
-  // A source of modules whose contents can be donated into the module being
-  // fuzzed.
-  const std::vector<fuzzerutil::ModuleSupplier>& donor_suppliers_;
-
-  // Random number generator to control decision making during fuzzing.
-  std::unique_ptr<RandomGenerator> random_generator_;
+  const MessageConsumer consumer_;
 
   // Determines whether all passes should be enabled, vs. having passes be
   // probabilistically enabled.
-  bool enable_all_passes_;
-
-  // Controls which type of RepeatedPassManager object to create.
-  RepeatedPassStrategy repeated_pass_strategy_;
+  const bool enable_all_passes_;
 
   // Determines whether the validator should be invoked after every fuzzer pass.
-  bool validate_after_each_fuzzer_pass_;
+  const bool validate_after_each_fuzzer_pass_;
 
   // Options to control validation.
-  spv_validator_options validator_options_;
+  const spv_validator_options validator_options_;
+
+  // Determines whether the probability of applying another fuzzer pass
+  // decreases as the number of applied transformations increases.
+  const bool continue_fuzzing_probabilistically_;
```

Comment:
I would add that this should be set to "true" if the fuzzer is being used in a stand-alone fashion to completely transform a module, and to "false" if the fuzzer is being used in an incremental manner (like we are doing in the tint fuzzer).

An alternative design would be to get rid of this field, and to invoke the "continue fuzzing probabilistically" behaviour only when "num_of_transformations_to_apply" is 0 - what do you think?

Reply:
Fixed in 5967f1f.